log/log.go

```diff
@@ -2,158 +2,62 @@ package log
 
 import (
 	"fmt"
-	"io"
 	"os"
-	"sync"
 
 	"github.com/ViaQ/logerr/kverrors"
 	"github.com/go-logr/logr"
 )
 
-// ErrUnknownLoggerType is returned when trying to perform a *Logger only function
-// that is incompatible with logr.Logger interface
-var ErrUnknownLoggerType = kverrors.New("unknown error type")
-
 var (
-	defaultOutput  io.Writer = os.Stdout
-	defautLogLevel           = 0
+	// ErrUnknownSinkType is returned when trying to perform a *Logger only function
+	// that is incompatible with logr.Logger interface
+	ErrUnknownSinkType = kverrors.New("unknown log sink type")
 
-	// logLevel sets the level at which you want logs to be displayed
-	// By default the verbosity is set to 0 and all logs that do not
-	// use V(...) will be printed. To increase logging verbosity
-	logLevel = defautLogLevel
+	defaultOutput = os.Stdout
 
-	mtx    sync.RWMutex
-	logger logr.Logger = NewLogger("", os.Stdout, 0, JSONEncoder{})
+	defaultLogger = logr.New(NewLogSink("", defaultOutput, 0, JSONEncoder{}))
 )
 
-// Init initializes the logger. This is required to use logging correctly
-// component is the name of the component being used to log messages.
-// Typically this is your application name.
-//
-// keyValuePairs are key/value pairs to be used with all logs in the future
-func Init(component string, keyValuePairs ...interface{}) {
-	InitWithOptions(component, nil, keyValuePairs...)
+// DefaultLogger creates a logger without any key value pairs
+func DefaultLogger() logr.Logger {
```

Comment:
Yeah, probably. I'm also wondering if I we should restructure some of this package's format to avoid this scenario:

`github.com/ViaQ/logerr/v2/kverrors`

(I might be doing something wrong here with this.)

Reply:
Yeah, the module name is different in v2, so the imports all need to be changed. Don't know if we need to add another `pkg` to the mix, though. In my mind, `pkg` is only useful for non-library projects to signal where the "reusable parts" are (the things for which the semantic versioning is applied). As this is a library everything should be reusable, except for `internal` packages.